lib/rules/hexadecimal-escape.ts

```diff
@@ -0,0 +1,125 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { Character } from "regexpp/ast"
+import type { RegExpContext } from "../utils"
+import {
+    defineRegexpVisitor,
+    createRule,
+    getEscapeSequenceKind,
+    EscapeSequenceKind,
+    toCharSetSource,
+    isHexadecimalEscape,
+} from "../utils"
+
+export default createRule("hexadecimal-escape", {
+    meta: {
+        docs: {
+            description: "enforce consistent usage of hexadecimal escape",
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [
+            {
+                enum: ["always", "never"], // default always
+            },
+        ],
+        messages: {
+            expectedHexEscape:
+                "Expected hexadecimal escape ('{{hexEscape}}'), but {{unexpectedKind}} escape ('{{rejectEscape}}') is used.",
+            unexpectedHexEscape:
+                "Unexpected hexadecimal escape ('{{hexEscape}}').",
+        },
+        type: "suggestion", // "problem",
+    },
+    create(context) {
+        const always = context.options[0] !== "never"
+
+        /**
+         * Verify for always
+         */
+        function verifyForAlways(
+            { node, getRegexpLocation, fixReplaceNode }: RegExpContext,
+            kind: EscapeSequenceKind,
+            cNode: Character,
+        ) {
+            if (kind === EscapeSequenceKind.hexadecimal) {
```

Comment:
I think we should explicitly check for Unicode escapes here. This will also make the octal and control check in `onCharacterEnter` unnecessary.

```suggestion
            if (kind !== EscapeSequenceKind.unicode && kind !== EscapeSequenceKind.unicodeCodePoint) {
```